modules/test/playwright/tests/account-admin-web/accountRole.spec.ts

```diff
@@ -0,0 +1,241 @@
+			account1.id,
+			[userAccount.emailAddress]
+		);
+
+		await apiHelpers.headlessAdminUser.assignUserToOrganizationByEmailAddress(
+			organization.id,
+			userAccount.emailAddress
+		);
+
+		await apiHelpers.jsonWebServicesUser.assignUsersToSite(
+			site.id,
+			userAccount.id
+		);
+
+		await performLogout(page);
+		await performLogin(page, userAccount.alternateName);
+
+		await page.goto(`/web/${site.name}/${layout.friendlyUrlPath}`);
+
+		const account1Cell = page.getByRole('cell', {
+			exact: true,
+			name: account1.name,
+		});
+
+		await expect(account1Cell).toBeVisible();
+		const tagName = await account1Cell.evaluate(
+			(element) => element.tagName
+		);
+		expect(tagName).not.toBe('a');
+
+		await expect(
+			page
+				.getByRole('cell', {exact: true, name: account2.name})
+				.getByText(account2.name)
+		).not.toBeVisible();
+
+		await performLogout(page);
+		await performLogin(page, 'test');
+
+		const accountManagerRole =
+			await apiHelpers.headlessAdminUser.getRoleByExternalReferenceCode(
+				'Account Manager'
+			);
+
+		await apiHelpers.jsonWebServicesUser.assignUsersToRole(
+			accountManagerRole.id,
+			[userAccount.id]
+		);
+
+		await performLogout(page);
+		await performLogin(page, userAccount.alternateName);
+
+		await page.goto(`/web/${site.name}/${layout.friendlyUrlPath}`);
+
+		const account1Link =
+			await accountManagementWidgetPage.accountsTableRowLink(
+				account1.name
+			);
+
+		await expect(account1Link).toBeVisible();
+		expect(account1Link.getAttribute('href')).not.toContain('null');
+
+		await expect(
+			await accountManagementWidgetPage.accountsTableRowLink(
+				account2.name
+			)
+		).toBeVisible();
+
+		for (const account of accounts) {
+			await (
+				await accountManagementWidgetPage.accountsTableRowLink(
+					account.name
+				)
+			).click();
+			await accountsPage.channelDefaultsTab.click();
+
+			await page.getByText('Billing Addresses').waitFor();
+
+			const entryHeaders = [
+				'Billing Addresses',
+				'Channel Account Managers',
+				'Currencies',
+				'Delivery Terms and Conditions',
+				'Discounts',
+				'Payment Methods',
+				'Payment Terms and Conditions',
+				'Price Lists',
+				'Shipping Addresses',
+				'Shipping Options',
+			];
+
+			for (const entryHeader of entryHeaders) {
+				const element = await page.$(`h3:has-text("${entryHeader}")`);
+
+				expect(await element.textContent()).toContain(entryHeader);
+			}
+
+			const entryNames = [
+				'Billing',
+				'Currencies',
+				'Delivery',
+				'Discounts',
+				'Payment',
+				'Price',
+				'Shipping',
+				'Users',
+			];
+
+			for (const entryName of entryNames) {
+				const element = await page.$(`[data-qa-id*='${entryName}']`);
+				const button = await element.$('button');
+
+				expect(
+					(await button.innerHTML()).includes(
+						'lexicon-icon lexicon-icon-plus'
+					)
+				).toBeTruthy();
+				expect(await element.textContent()).toContain(
+					'No Results Found'
+				);
```

Comment:
same here... we (also you) always used something like toHaveText or toContainText ...